sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,933 @@
+ */
+export type TopicResponse = Topic & {
+  /**
+   * The underlying HTTP response.
+   */
+  _response: HttpOperationResponse;
+};
+
+/**
+ * Create Topic response
+ */
+export type CreateTopicResponse = TopicResponse;
+
+/**
+ * Get Topic response
+ */
+export type GetTopicResponse = TopicResponse;
+
+/**
+ * Update Topic response
+ */
+export type UpdateTopicResponse = TopicResponse;
+
+/**
+ * Delete Topic response
+ */
+export type DeleteTopicResponse = TopicResponse;
+
+/**
+ * Represents result of list operation on topics.
+ */
+export interface ListTopicsResponse extends Array<Topic> {
+  /**
+   * The underlying HTTP response.
+   */
+  _response: HttpOperationResponse;
+}
+
+/**
+ * Represents result of create, get, update and delete operations on subscription.
+ */
+export type SubscriptionResponse = Subscription & {
+  /**
+   * The underlying HTTP response.
+   */
+  _response: HttpOperationResponse;
+};
+
+/**
+ * Create Subscription response
+ */
+export type CreateSubscriptionResponse = SubscriptionResponse;
+
+/**
+ * Get Subscription response
+ */
+export type GetSubscriptionResponse = SubscriptionResponse;
+
+/**
+ * Update Subscription response
+ */
+export type UpdateSubscriptionResponse = SubscriptionResponse;
+
+/**
+ * Delete Subscription response
+ */
+export type DeleteSubscriptionResponse = SubscriptionResponse;
+
+/**
+ * Represents result of list operation on subscriptions.
+ */
+export interface ListSubscriptionsResponse extends Array<Subscription> {
+  /**
+   * The underlying HTTP response.
+   */
+  _response: HttpOperationResponse;
+}
+
+/**
+ * Represents result of create, get, update and delete operations on rule.
+ */
+export type RuleResponse = Rule & {
+  /**
+   * The underlying HTTP response.
+   */
+  _response: HttpOperationResponse;
+};
+
+/**
+ * Create Rule response
+ */
+export type CreateRuleResponse = RuleResponse;
+
+/**
+ * Get Rule response
+ */
+export type GetRuleResponse = RuleResponse;
+
+/**
+ * Update Rule response
+ */
+export type UpdateRuleResponse = RuleResponse;
+
+/**
+ * Delete Rule response
+ */
+export type DeleteRuleResponse = RuleResponse;
+
+/**
+ * Represents result of list operation on rules.
+ */
+export interface ListRulesResponse extends Array<Rule> {
+  /**
+   * The underlying HTTP response.
+   */
+  _response: HttpOperationResponse;
+}
+
+/**
+ * All operations return promises that resolve to an object that has the relevant output.
+ * These objects also have a property called `_response` that you can use if you want to
+ * access the direct response from the service before.
+ */
+export class ServiceBusAtomManagementClient extends ServiceClient {
+  private endpoint: string;
+
+  private queueResourceSerializer: AtomXmlSerializer;
+  private topicResourceSerializer: AtomXmlSerializer;
+  private subscriptionResourceSerializer: AtomXmlSerializer;
+  private ruleResourceSerializer: AtomXmlSerializer;
```

Comment:
Can this only use one serializer, or are they differently configured?

Reply:
There's one serializer for each entity type

Reply:
Can we have a single one for all entity types?

Reply:
Each of them deal with different properties that can be set and read. So, am ok with having different serializers for each entity. The one optimization here that can be done is that the serializers can be static classes as they don't maintain state. That way we don't have to instantiate them and maintain them as private members this way. But, I wouldn't have that as a blocker for this PR. It can be a future optimization.